src/components/SmartLabel.brs

```diff
@@ -185,28 +188,31 @@ function makeFontSetting(name, size, color = "#000000")
     size: size
     color: color
   }
-end function 
+end function
 
-function udpateFontSetting(key, name, size, color = "#000000")
+function updateFontSetting(key, name, size, color = "#000000")
+  style = key.replace("Settings", "")
   if m.settingsKeys[key] <> invalid
-    m[style] = makeFontSetting(name, size, color)
+    m[style] = getFont(name, size)
+    m[style + "Color"] = color
   else
     ? "unknown style " ;style
   end if
 end function
 
-function onAllFontSettingsChange(settings)
-  for each key in settings
-    setting = settings[key]
-    
-    if (type(setting) = "roAssociativeArray")
-      if m.settingsKeys[key] <> invalid
-        m[style] = setting
+function onAllFontSettingsChange(event as Object)
+  settings = event.getData()
+  for each style in settings
+    setting = settings[style]
+    if type(setting) = "roAssociativeArray"
+      if m.settingsKeys[style + "Settings"] <> invalid
+        m[style] = getFont(setting.name, setting.size)
```

Comment:
this is wrong. m[style] contains an aa style - here we've changed it to contain the font. Please read the xml comment that goes with the `allFontSettings` field. i.e. 
```
    <!-- convenience, set as: 
        { "KEY": {
          name: name of font (pkg or font: location)
          size: font size
        color: color
      } 
    where key is one of:
     - fontSettings
     - header1FontSettings
     - header2FontSettings
     - header3FontSettings
     - italicFontSettings
     - boldFontSettings
    -->
    <field
        id='allFontSettings'
        type='assocarray' />
```